### ISSUE TITLE
Caisse SQL - poor design, should be 'enhanced'

### ISSUE THREAD

**Issue by AVHuntley** [afs1jes/Clarins-CA-IR-Migration#56]
![Image](https://user-images.githubusercontent.com/88676502/217946240-e6320dd7-a789-4795-a9bf-5bb8edbede29.png)

**Comment by A3Huntley** [comment]
As far as I can tell, this is now working identically on Dev as on Prod. IR build works.

I don't know what the template is supposed to do, but I agree that it doesn't seem functional as is.

**Comment by FrederickGiroux** [comment]
The report is finished and working well. Ready to migrate

**Comment by A3Huntley** [comment]
Great! I pulled this template from prod then ran it through the migration process. Seems to be working for me on Dev now.

You might want to consider forbidding non-leaf members in the selectables. Non-leaves can mess up the SQL queries because they don't explicitly exist in the SQL tables. (e.g., The TIMENAME column doesn't contain "Full Year". So a query that contains WHERE TIMENAME = 'Full Year' will find 0 records.)

It's also possible to use a formatting row to hold your formulas, instead of copying them down the page. It involves adding an [alias column](https://www.w3schools.com/sql/sql_alias.asp) to the SQL query, then pasting the formula in the associated formatting row. Since you've been learning SQL, I can show you how, if you're interested! Completely optional though.

**Comment by FrederickGiroux** [comment]
Good point for the TimeName, in that case i know the user of that report and the risk is minimal. 

Definitly interested to learn that, but i think i'll put all my ''A3 time'' into reviewing for now. I would be really happy to look into that once we finish the dev site. Also i've have been struggling with negative numbers combined with the between function haha. I'll make a list of things we can look into in the future.

Otherwise the SQL's sheets seem to work fine, the first sheet doesn't work and i cannot put my finger on why. It should look like the following picture if you select jan 2023

**Comment by FrederickGiroux** [comment]
![Image](https://user-images.githubusercontent.com/124090756/235787473-8713639e-77a5-4ba8-ba8d-f577f767f342.png)

**Comment by A3Huntley** [comment]
Good catch. Turns out that the server-build process was returning the values from your SQL grids in text form, so Excel didn't recognize them as number values.

We made an adjustment to the A3 add-in that should address this issue. This report seems to function correctly now. Can you try again and let us know if it's working for you as well?
